### PR TITLE
Client can signup and create assignments

### PR DESCRIPTION
**Pull request by ctl-lab**
Screenshot 2020-11-07 at 13 40 40
[PT](https://www.pivotaltracker.com/story/show/175525251) 

## This PR contains
- Feature tests with Detox
- Ability for client to signup
- Ability for client to create assignments
- Redux setup
- AsyncStorage setup

Error1

### CODE REVIEW

File: e2e/clientCanCreateAssignment.e2e.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    await element(by.id("budgetInput")).typeText("500");
+    await element(by.id("skills-1")).tap();
+    await expect(element(by.id("publishButton"))).toBeVisible();
+    await element(by.id("publishButton")).tap();
+    await expect(element(by.id("successfullyCreatedMessage"))).toBeVisible();
+  });
+  it("unsuccessfully", async () => {
+    const randomize = Math.floor(Math.random() * 10000);
+    await expect(element(by.id("wantToPublishButton"))).toBeVisible();
+    await element(by.id("wantToPublishButton")).tap();
```
Comment (SaraFrantzen) [inline e2e/clientCanCreateAssignment.e2e.js]: Just explain in title how it is unsuccessful. 
Unsuccessful with missing *input

File: e2e/clientCanSignUp.e2e.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    await element(by.id("mainContainer")).tap();
+    await expect(element(by.id("submitButton"))).toBeVisible();
+    await element(by.id("submitButton")).tap();
+    await expect(element(by.id("welcomeMessage"))).toBeVisible();
+    await expect(element(by.id("createAssignmentButton"))).toBeVisible();
+  });
+  it("unsuccessfully", async () => {
+    await expect(element(by.id("wantToPublishButton"))).toBeVisible();
+    await element(by.id("wantToPublishButton")).tap();
+    await element(by.id("emailInput")).tap();
```
Comment (SaraFrantzen) [inline e2e/clientCanSignUp.e2e.js]: same here, unsuccessfully with non matching password
and then change the password conf input

File: index.js [hunk 3]
```diff
@@ -2,7 +2,21 @@ import React from "react";
 import { registerRootComponent } from "expo";
 import App from "./App";
 import axios from "axios";
+import { Provider } from "react-redux";
+import configureStore from "./src/state/store/configureStore";
 
 axios.defaults.baseURL = "https://develup-2020.herokuapp.com/api";
+// axios.defaults.baseURL = "http://localhost:3000/api";
+//  axios.defaults.baseURL = "http://cbf80599e6ab.ngrok.io/api";
```
Comment (SaraFrantzen) [inline index.js]: delete this for merge

File: src/components/ClientSignUp.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    <Container>
+      <Content testID="mainContainer">
+        <Form>
+          <Item floatingLabel>
+            <Label testID="emailInput">Email</Label>
+            <Input
+              testID="emailField"
+              onChangeText={(text) => setEmail(text)}
+            />
+          </Item>
```
Comment (SaraFrantzen) [inline src/components/ClientSignUp.js]: change the name's of testID, r.61 should be emailLabel and r.63 should be emailInput. Make sure to change in the test

File: src/components/Checkbox.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { CheckBox } from "native-base";
+import { StyleSheet } from "react-native";
+const Checkbox = (props) => {
+  return (
+    <>
+      <CheckBox style={styles.checkbox} checked={props.isChecked} />
```
Comment (FacundoOs) [inline src/components/Checkbox.js]: A component for only one line??

**Review by FacundoOs — APPROVED** [review]

**Pull request merged**